src/hypertable.c

```diff
@@ -812,6 +812,16 @@ table_has_tuples(Oid table_relid, Snapshot snapshot, LOCKMODE lockmode)
 	return hastuples;
 }
 
+static bool
+table_has_replica_identity(Oid table_relid, LOCKMODE lockmode)
```

Comment:
Maybe these checks should be syscache "scans" instead, like `get_rel_name`, `get_rel_persistence` in `lsyscache.c`. Seems `heap_open` is heavy-weight unless heap already open of course.